props/saber_BC_buttons.h

```diff
@@ -382,40 +1148,252 @@ EFFECT(volup);      // for increse volume
+
+  void prev() override {
+    highlighted_blade_.Stop(current_blade());
+    SPEC::MenuBase::prev();
+    highlighted_blade_.Start(current_blade());
+    PVLOG_NORMAL << "** Highlighting blade: " << current_blade() << "\n";
+  }
+
+  void say() override {
+    mode::getSL<SPEC>()->SayBlade();
+    mode::getSL<SPEC>()->SayWhole(current_blade());
+  }
+
+  void select() override {
+    // Set the current blade to send and push to ChangeBladeLengthMode
+    mode::menu_current_blade = current_blade();
+    highlighted_blade_.Stop(current_blade());
+    pushMode<typename SPEC::ChangeBladeLengthMode>();
+  }
+};
+
+template<class SPEC>
+// struct BCChangeBladeLengthBlade1 : public SPEC::ChangeBladeLengthMode {
+struct BCChangeBladeLengthBlade1 : public mode::ChangeBladeLengthBlade1<SPEC> {
+  // virtual int blade() { return mode::menu_current_blade; }
+
+  int steps_per_revolution() override {
+    return 30;  // adjust for sensitivity
+  }
+  void select() override {
+    prop_SetBladeLength(this->blade(), this->getLength());
+    prop_SaveState();
+    mode::getSL<SPEC>()->SaySave();
+    popMode();
+  }
+  void update() override {
+    hybrid_font.PlayPolyphonic(&SFX_volup);
+    this->say_time_ = Cyclint<uint32_t>(millis()) + (uint32_t)(SaberBase::sound_length * 1000) + 300;
+    if (!this->say_time_) this->say_time_ += 1;
+    this->fadeout(SaberBase::sound_length);
+  }
+};
+#endif  // DYNAMIC_BLADE_LENGTH
+
+template<class SPEC>
+struct BCMenuSpec {
+  typedef BCVolumeMode<SPEC> BCVolumeMenu;
+#ifdef DYNAMIC_BLADE_LENGTH
+  typedef BCSelectBladeMode<SPEC> BCSelectBladeMenu;
+  typedef mode::ShowLengthStyle<SPEC> ShowLengthStyle;
+  typedef BCChangeBladeLengthBlade1<SPEC> ChangeBladeLengthMode;
+#endif  // DYNAMIC_BLADE_LENGTH
+  typedef mode::SelectCancelMode SelectCancelMode;
+  typedef mode::SteppedMode<SPEC> SteppedMode;
+  typedef mode::SteppedModeBase<SPEC> SteppedModeBase;
+  typedef mode::MenuBase<SPEC> MenuBase;
+  typedef SoundLibraryV2 SoundLibrary;
+};
+
+#ifndef MENU_SPEC_TEMPLATE
+  void Setup() {
+    MKSPEC<BCMenuSpec>::SoundLibrary::init();
+  }
+#endif
 
 class DelayTimer {
 public:
-    DelayTimer() : triggered_(false), trigger_time_(0), duration_(0) {}
+  DelayTimer() : triggered_(false), trigger_time_(0), duration_(0) {}
 
-    void trigger(uint32_t duration) {
-        triggered_ = true;
-        trigger_time_ = millis();
-        duration_ = duration;
-    }
+  void trigger(uint32_t duration) {
+    triggered_ = true;
+    trigger_time_ = millis();
+    duration_ = duration;
+  }
 
-    bool timerCheck() {
-        if (!triggered_) return false;
-        if (millis() - trigger_time_ > duration_) {
-            triggered_ = false;  // Reset the timer flag
-            return true;  // Timer has elapsed
-        }
-        return false;  // Timer is still running
+  void stopTimer() {
+    PVLOG_DEBUG << "** STOPPING timer.\n";
+    triggered_ = false;
+  }
+
+  bool isTimerExpired() {
+    if (!triggered_) return false;
```

Comment:
This is super confusing, because isTimerExpired and isTimerRunning sounds very similar, and it seems like they should do basically the same thing. However, isTimerRunning() returns true every time you call it until the timer has expired, but isTimerExpired() only returns true once.

What's worse is that if you call both isTimerRunning and isTImerExpired on repeatedly on the same timer, isTimerExpired might never trigger, because the one time it returned true actually occurred when it was called from isTimerRunning().

Reply:
They shouldn't ever get called repeatedly, but after moving volume menu stuff to modes, the only thing left calling isTimerRunning() is DoQuote(), which can just check isTimer NOT Expired instead.
Adjusted and tested. Better now?

Reply:
Ok so this needed additional work because it's a little different than the mute and beep timers. Those run a timer to preempt another action, whereas to just prevent overlap for quotes (and force too now) we need to play the sound once to get a duration before starting a timer. So, to bypass checking on the first occurence, I added boolean flag that gets set true once a timer has been used.
I also renamed quote delay timer to overlap_delay_timer since it now has multiple use cases, as well as clarified the other timer names to specify they are used for delaying things.